File: cgi-bin/test_cgi.php
<?php
// Get the POST data
$postData = $_POST['data'];
$mail = $_POST['mail'];

// Output HTTP headers
header("Content-Type: text/html");

// Output the HTML response
echo "<html><body>";
echo "<h1>CGI Test (PHP)</h1>";

if (!empty($postData)) {
    echo "<p>Received POST data: " . htmlspecialchars($postData) . "</p>";
    echo "<p>Received POST mail: " . htmlspecialchars($mail) . "</p>";
} else {
    echo "<p>No POST data received.</p>";
}

echo "</body></html>";
?>

File: cgi-bin/test_cgi.py
#!/usr/bin/env python3

import cgi

print("Content-Type: text/html\n")
print("<html><body>")
print("<h1>CGI Test</h1>")

form = cgi.FieldStorage()

if "data" in form:
    print("<p>Received POST data: {}</p>".format(form["data"].value))
else:
    print("<p>No POST data received.</p>")

print("</body></html>")
